homeassistant/components/wiffi/config_flow.py

```diff
@@ -38,6 +38,8 @@ async def async_step_user(self, user_input=None):
             return self._async_show_form()
 
         # received input from form or configuration.yaml
+        await self.async_set_unique_id(user_input[CONF_PORT])
```

Comment:
A port address is normally not allowed as unique_id since it may not be stable. Is this a different case?

https://developers.home-assistant.io/docs/config_entries_config_flow_handler#unique-id-requirements

Reply:
From my point of view, this is no different from a hostname (that is also acceptable for local devices). The port is a stable, non-user-changeable fixed ID, with the sole purpose to prevent duplicate entries for the same port to be created.

Reply:
I'm asking if it is stable. I'd like that confirmed.

Reading the docs page it sounds as the user configures the port on both sides.
https://www.home-assistant.io/integrations/wiffi/

Reply:
This is the port being configured on the Home Assistant end, which is opened. That is all on our end, and in the current configuration flow.

There is no logic that changes this port on this end either.

Reply:
Since the user can change the port for the same device, I don't consider it stable.

Reply:
A user can also change a hostname or IP address of the same device for many integrations on local networks... I'm not sure if I follow your reasoning?

Reply:
We don't allow ip address as unique id. We don't allow hostname if the user can it. Hostname is only allowed if it contains one of the acceptable sources, then that portion can be used.

Reply:
🤷  I don't see any harm in this one. But if you disagree I suggest you either fix or revert it.

Reply:
We should use this helper instead:
https://github.com/home-assistant/core/blob/52ca06c750539164764793ed6edd0d8c6853dd24/homeassistant/config_entries.py#L1200-L1204

@mampfes please address it in a new PR. Thanks!

Reply:
If you think that should be done @MartinHjelmare, please revert this PR to prevent the ID being set for users and preventing it actually ending up in beta or stable.

Reply:
There's no rush yet. Let's wait a bit for @mampfes to respond.

Reply:
The port is used to open a TCP server socket. On the client device side, the hostname and the socket port has to be configured, too. Currently, the integration does not allow to change the port, therefore - currently - the port is unique.

`_async_abort_entries_match` should have the same outcome, therefore I think it is ok to use that one. I can provide a PR.

Reply:
Great! Please make a PR. 👍